components/style/properties/properties.mako.rs

```diff
@@ -1091,6 +1098,10 @@ impl PropertyDeclaration {
                     match input.try(|i| CSSWideKeyword::parse(context, i)) {
                         Ok(CSSWideKeyword::InheritKeyword) => {
                             % for sub_property in shorthand.sub_properties:
+                                if properties_seen.get_${property.ident}() {
```

Comment:
nit: you could write this in a more compact way (here and everywhere else) as follows:

```
*possibly_duplicated |= properties_seen.get_${property.ident}();
properties_seen.set_${property.ident}();
```